### PR TITLE
ApiKey docs

### PR DESCRIPTION
**Pull request by lgadban**
add more detail to docs regarding apikey auth

### CODE REVIEW

**Comment by lgadban** [comment]
Should close https://github.com/solo-io/solo-projects/issues/1683

**Comment by lgadban** [comment]
@rickducott -- Since this has changes to the protobufs and gen'd code, does it need to go through CI?

File: docs/content/guides/security/auth/apikey_auth/_index.md [hunk 1]
```diff
@@ -23,11 +23,13 @@ rotation is up to the user, thus the security of the routes is up to the user.
...
 
-1. you can specify a **label selector** that matches one or more labelled API key secrets (this is the preferred option), or
-1. you can **explicitly reference** a set of secrets by their identifier (namespace and name).
+1. You can specify a **label selector** that matches one or more labelled API key secrets (this is the preferred option), or
+1. You can **explicitly reference** a set of secrets by their identifier (namespace and name).
 
 When Gloo matches a request to a route secured with API keys, it looks for a valid API key in the `api-key` header. If the header is not present, or if the API key it contains does not match one of the API keys in the secrets referenced on the Virtual Service, Gloo will deny the request and return a 401 response to the downstream client.
 
+Internally, Gloo will generate a mapping of API keys to _user identities_ for all API keys present in the system. The _user identity_ for a given API key is the name of the `Secret` which contains the API key. The _user identity_ will be added to the request as a header, `x-user-id` by default, which can be utilized in subsequent filters. This header is then sanitized from the response.
```
Comment (kdorosh) [inline docs/content/guides/security/auth/apikey_auth/_index.md:31]: > which can be utilized in subsequent filters.

perhaps a good place to link to the default filter ordering, https://github.com/solo-io/gloo/blob/master/projects/gloo/pkg/plugins/plugin_interface.go#L187-L198

**Comment by kdorosh** [comment]
/skip-ci, only docs changes. so long as codegen passes (it did) then no need to run ci

File: docs/content/guides/security/auth/apikey_auth/_index.md (hunk 1 above)
Comment (lgadban) [reply docs/content/guides/security/auth/apikey_auth/_index.md]: I added a link to this and some more explanation. Let me know if you think this paragraph is now getting too verbose.

**Pull request merged**